rudder-lang/tools/cfjson_tester

```diff
@@ -133,54 +161,35 @@ def remove_wilful_entries(diff):
   ## del superfluous if true equivalent condition
   deep_del(diff, ['bundles', 'promiseTypes', 'contexts', 'promises', 'attributes', 'rval', 'arguments'], {'type': 'string', 'value': 'any'}, '$delete')
 
-def compare_cf(first, second):
-  j1 = load_cf(first)
-  j2 = load_cf(second)
-  diff = jsondiff.diff(j1, j2, syntax="symmetric", marshal=True)
-  remove_wilful_entries(diff)
-  empty_dict_cleaner(diff)
-  if len(diff) != 0:
-    print(json.dumps(diff, indent=2))
-
-def load_cf(file_name):
-  # Check file exists
-  if not os.path.exists(file_name):
-    print("No such file: " + file_name)
-    exit(1)
-
-  env = os.environ.copy()
-  env['RES_OPTIONS'] = 'attempts:0'
-  out = check_output([ncf.CFENGINE_PATH, "-pjson", "-f", file_name], env=env)
-  try:
-    promises = json.loads(out)
-  except Exception as e:
-    print("An error occured while parsing technique '"+file_name+"'", cause = e)
-    exit(1)
-  return promises
-
-def compare_cf(first, second):
-  j1 = load_cf(first)
-  j2 = load_cf(second)
-  diff = jsondiff.diff(j1, j2, syntax="symmetric", marshal=True)
-  print(json.dumps(diff, indent=2))
-
-def get_config(path):
-  cfgpath = 'tools/rudderc.conf'
-  if path:
-    cfgpath = path
-  cfg_full_path = os.getcwd() + '/' + cfgpath
-  if not os.path.exists(cfg_full_path):
-    print("No such file: " + cfg_full_path)
+def get_config(cfgpath):
+  if cfgpath is None:
+    cfgpath='/opt/rudder/etc/rudderc.conf'
+  if not os.path.exists(cfgpath):
+    print_json(cfg_path, 'No config file', True)
     exit(1)
 
   cfg = configparser.ConfigParser()
-  cfg.read(cfg_full_path)
+  cfg.read(cfgpath)
   return cfg
 
 if __name__ == "__main__":
-  arguments = docopt.docopt(__doc__)
+  import docopt
```

Comment:
if it is not installed this will fail, i think you meant to put this in the try catch

Reply:
you can make this work everywhere by just adding rudder python path all the time
 sys.path.append("/opt/rudder/share/python/")

Reply:
because we always have docopt installed ?

Reply:
Ok fixed. Still kept the try execpt as dev env might not have `/opt/rudder/share/python/docopt`

Reply:
you don't need it, you may want to put a try around the import itself to provide an error message to the developper, but the developper should already know how to fix this.
The rest of your try/except is just duplicate code

Reply:
fixed